src/v/cluster/config_frontend.cc

```diff
@@ -15,6 +15,7 @@
 #include "cluster/controller_service.h"
 #include "cluster/partition_leaders_table.h"
 
+#include <ranges>
```

Comment:
>cluster/config_frontend: convert aliases to primary names before committing to the log

Can you elaborate on this a bit? There seems to be some nuanced logic around versions and it would be great to explain the rationale. Like how does the controller handle names that weren't normalized, etc?